### ISSUE TITLE
Fails to start from launcher on SFOS 4.0.1

### ISSUE THREAD

**Issue by Olf0** [sailfishos-chum/sailfishos-chum-gui#111]
The SailfishOS:Chum GUI application 0.4.1-1.5.1.jolla fails to start when tapped at the launcher on SFOS 4.0.1 (Xperia XA2+).  Always, i.e. 100% reproducible.  It fails immediately, i.e., nothing really happens, as if tapping on its icon has no action assigned.

It was downloaded from https://chumrpm.netlify.app and installed via file-manager.

It works when started at the command line, but emits a warning (repeatedly, when used):
`[W] unknown:479 - file:///usr/lib/qt5/qml/Sailfish/Silica/private/PulleyMenuBase.qml:479:13: QML State: Binding loop detected for property "when"`

I also observe a couple of other errors, which are likely unrelated, when looking at Patchmanager:
1. Patchmanager is not displayed as an "application", although it is not part of the groups "Library" or "Development".  Patchmanager is solely part of the group "Other", but only shown when listing "all packages", in contrast to e.g., FoilAuth, which also is solely part of the group "Other"!?!  This happens regardless of entering the category "Other" first or just the top level entry "Applications".
2. I have Patchmanager `3.2.0-1` installed from a different source.  I enabled the SailfishOS:Chum testing repository, where Patchmanager `3.2.1-1.9.1.jolla` is available, but it is not found as a update, although I can find it by searching (see point 1).  This might be intended behaviour (thus O.K.) in order not to interact with packages which were installed from other sources, see also point 3.
3. Patchmanager *is* listed when entering the top level entry "Installed packages".  The Patchmanager entry shows `3.2.0-1` as installed and `3.2.1-1.9.1.jolla` as available.  Still, I cannot update it there either, only "Uninstall".
    This might be intended behaviour (thus O.K.) in order not to interact with packages which were installed from other sources, see also point 2.
4. In the Releases page of Patchmanager, [all releases at GitHub](https://github.com/sailfishos-patches/patchmanager/releases?page=2) are displayed, but all older than `2.3.0-dev` with a completely blank entry (i.e., no tag / version number), except for the (separately extracted and displayed) date the release tag was set.
    When entering the details page of such a blank entry, a completely empty page is shown (except for the release tag date); really nothing: no version number, no comment, nothing, even though all this exists at GitHub!
5. I have to enter my security code each time the SailfishOS:Chum repository is refreshed, which includes every start of the SailfishOS:Chum GUI application.  This is very inconvenient.
    *Edit:* I just found out, that the SailfishOS:Chum GUI application seems to work fine, when I dismiss this dialogue by hitting "Cancel" or tapping elsewhere on the screen.  I wonder what spawning it is good for.

P.S.:
6. Slava @monich's counter app has an SVG icon (which is an unsupported file format) referenced in its SailfishOS:Chum metadata, which also shows when started at the command line.

P.P.S.: Lastly I gave up, downloaded Patchmanager 3.2.1 from the SailfishOS:Chum testing repository in a web-browser and installed it manually.

**Comment by monich** [comment]
> 6. Slava @monich's counter app has an SVG icon (which is an unsupported file format) referenced in its SailfishOS:Chum metadata, which also shows when started at the command line.

Version 1.0.23 of harbour-counter should fix the chum icon. Let me know if the problem persists.

**Comment by Olf0** [comment]
> > 6. Slava @monich's counter app has an SVG icon (which is an unsupported file format) referenced in its SailfishOS:Chum metadata, which also shows when started at the command line.
> 
> Version 1.0.23 of harbour-counter should fix the chum icon. Let me know if the problem persists.

I don't think so, because it is you.  Sorry, too busy, overloaded & lazy to check, because I know you are ultra-reliable. :grinning:
(The same would apply to such a statement from @rinigus.)

**Comment by Olf0** [comment]
And I hit

7. For some reason the the whole metadata for SailfishOS:Chum is displayed as part of Patchmanager's description in the SailfishOS:Chum GUI application.
    I try to alleviate this by [Patchmanager commit 9084f3e](https://github.com/sailfishos-patches/patchmanager/pull/273/commits/9084f3eaaa3dfb81e1a677319d5fea3ebb527493) (as part of [Patchmanager PR 273](https://github.com/sailfishos-patches/patchmanager/pull/273)) but doubt that it makes a difference.
    
    So I looked at [the spec file of Slava @monich's counter app](https://github.com/monich/harbour-counter/blob/master/rpm/harbour-counter.spec), to see if there are structural any differences: I am unable to spot any.
<br />

P.S.: Slava, now that I was looking into the spec file of your counter app for a different reason, I still see [an SVG icon referenced in the SailfishOS:Chum metadata](https://github.com/monich/harbour-counter/blob/master/rpm/harbour-counter.spec#L33).  Well, maybe you are not publishing from the master branch.

**Comment by monich** [comment]
> Slava @monich, now that I was looking into the spec file of your counter app for a different reason, I still see [an SVG icon referenced in the SailfishOS:Chum metadata](https://github.com/monich/harbour-counter/blob/master/rpm/harbour-counter.spec#L33) there. Well, maybe you are not publishing from the master branch.

SVG is should be, that's an image format supported by Qt, assuming that qt5-qtsvg-plugin-imageformat-svg package is installed. The problem with the previous version was that URL was pointing to an HTML page, this one actually returns SVG. The SVG can then be rendered by the client to the required pixel resolution without loosing the quality. That was the idea, I thought. But since it wasn't my idea, I'm not actually sure!

**Comment by piggz** [comment]
I wonder if it is all the comment lines.  I think the parsing is based on "looking for the last paragraph" and those comment lines maybe treated as blank lines. @rinigus  ?

**Comment by rinigus** [comment]
Trying to catch up with the issue. @Olf0: please don't mix the issues. Sounds like all discussion is regarding Patchmanager metadata and not what's in the header.

Looking into Patchmanager description from `zypper`:

```
Information for package patchmanager:
-------------------------------------
Repository     : sailfishos-chum
Name           : patchmanager
Version        : 3.2.1-1.5.1.jolla
Arch           : aarch64
Vendor         : chum
Installed Size : 1.3 MiB
Installed      : No
Status         : not installed
Source package : patchmanager-3.2.1-1.5.1.jolla.src
Summary        : Allows to manage Patches for Sailfish OS
Description    : 
    Patchmanager is a tool for transparently modifying installed files by the patch
    utility and for managing the special patch files ("Patches") for doing so.
    Since version 3.0 it does not modify original files, but alters their content
    when they are loaded into RAM to be run.



    PackageName: Patchmanager
    Type: desktop-application
    Categories:
     - System
     - Settings
     - Utility
    Custom:
      Repo: https://github.com/sailfishos-patches/patchmanager
    Icon: https://raw.githubusercontent.com/sailfishos-patches/patchmanager/master/src/plugin/icon-m-patchmanager.png
    Url:
      Homepage: https://openrepos.net/content/patchmanager/patchmanager


      Help: https://github.com/sailfishos-patches/patchmanager/blob/master/README.md

      Bugtracker: https://forum.sailfishos.org/t/bugs-in-patchmanager-3-1-0/8552
```

So, all comments are replaced by newlines as shown in the description. Chum GUI would take the last paragraph of text (as indicated by newline without text) and parse it. So, instead of your description, you get 
```
      Bugtracker: https://forum.sailfishos.org/t/bugs-in-patchmanager-3-1-0/8552
```
to parse. Which it fails (it is not even under "Url" subsection). 

@Olf0: Please drop the comments and, if you want to keep them, move out of Chum metadata section. 

As for Gui no starting. Looks like we should change the cutoff and set it higher. Try to make the replacement in .desktop file. As you already know where it is (you have found it earlier), I presume I don't need to specify location. :)

**Comment by Olf0** [comment]
Sorry, Slava @monich, I thought and wrote nonsense.  I just did too many things at once and became confused.
 
> SVG […] should be […] an image format supported …

It sure is, and (a few weeks ago) [it was me who documented that](https://github.com/sailfishos-chum/main/blame/main/Metadata.md#L44).

Plus I checked in practice, and all is fine (WRT the SVG icon of your counter app).

**Comment by Olf0** [comment]
> …
> So, all comments are replaced by newlines as shown in the description. Chum GUI would take the last paragraph of text (as indicated by newline without text) and parse it.
> …
> @Olf0: Please drop the comments and, if you want to keep them, move out of Chum metadata section.

Well, I did not introduce them: I will discuss among the Patchmanager maintainers where to move them.

Good to know that the SailfisOS:Chum GUI app looks for the last uninterrupted block in the `%description` section.

> As for Gui no starting. Looks like we should change the cutoff and set it higher. Try to make the replacement in .desktop file. As you already know where it is (you have found it earlier), I presume I don't need to specify location. :)

Well, I now know where it is, but have no idea what "*change the cutoff and set it higher. Try to make the replacement in .desktop file.*" means!

**Comment by rinigus** [comment]
My bad - please see line 

https://github.com/sailfishos-chum/sailfishos-chum-gui/blob/main/rpm/sailfishos-chum-gui.spec#L77

You would have to replace `silica-qt5` type to `generic` in desktop file.

**Comment by Olf0** [comment]
> You would have to replace silica-qt5 type to generic in desktop file.

Works fine, as expected (now that I understood what this was all about).

> My bad - please see line
https://github.com/sailfishos-chum/sailfishos-chum-gui/blob/main/rpm/sailfishos-chum-gui.spec#L77

Ah, this was meant with "cutoff": Should be at least `41000` then.

I will try to remember to check this when I update this XA2+, but there is one thing I still intend to evaluate on SFOS 4.0.1.

**Comment by Olf0** [comment]
@rinigus, do you really want me to split the remaining points 1 to 5 into separate issues, now the at the primary issue plus points 6 and 7 are resolved.  I was hoping that you may dismiss some of them as "please look twice", "please think twice" etc., so I can spare the effort. :wink:

**Comment by rinigus** [comment]
And coming back to question whether to file separately - let's see if we can fix them before.

**Comment by Olf0** [comment]
> Those Patchmanager related bugs, have those been fixed now with your patch to it?

Thank you, that is what I meant.  No, I have considered that as a consequence of the messed up metadata for SailfisOS:Chum; but that makes sense.  I do not think I will be able to go through the tagging at GH, triggering OBS and testing cycle before next weekend.

Practical question: Can I create an arbitrary tag at GitHub (say, 3.2.1-567), put that in the `_service` and all will work, while the spec file still says "Version: 3.2.1  Release: 1".

I only doubt that point 4 is influenced by this.  But I will check that then, too.

> Regarding point 5: is it if launched from Lipstick or from terminal? Could be some interference with packagemanager

All above points were documented, when only starting from the command line worked for me.

**Comment by rinigus** [comment]
> Practical question: Can I create a tag with an arbitrary release-version at GitHub (say, 3.2.1-567), put that in the _service and all will work, while the spec file still says "Version: 3.2.1 Release: 1"?

Yes, you can do that. In general, we don't recommend to tag in format a.b.c-d - better use a.b.c.d . Reason is simple: OBS will change -d part incrementally to distinguish the builds. In practice, it will rewrite Release section as well, according to your git tag.

So, feel free to tag again and trigger the build without changing SPEC

Re point 4: you would have to file it as a separate issue then. 

Re 5: with .desktop fixed and starting from GUI - would you still need to enter PIN?

**Comment by Olf0** [comment]
> Yes, you can do that. In general, we don't recommend to tag in format a.b.c-d - better use a.b.c.d . Reason is simple: OBS will change -d part incrementally to distinguish the builds. In practice, it will rewrite Release section as well, according to your git tag.

O.K., so I went from the `3.2.1` release number to the `3.2.1.1` test release number.  This worked fine at SailfishOS:Chum testing, exactly as you denoted, but I missed to consider the GitHub-actions build runner, which built another `3.2.1-1`.  Well, not your business, but for the next test release I will try to remember altering the spec file in a testing branch and tagging the testing release there with a corresponding `<previous three field release number>.<number of the test release>` version string (and holding the release version constantly at `-1`, as OBS is assigning a new release version any way, as you indicated). 

> So, feel free to tag again and trigger the build without changing SPEC

Did that and points 1 to 3 are indeed resolved. :smile:

> Re point 4: you would have to file it as a separate issue then.

I will, because it stayed, as expected.

> Re 5: with .desktop fixed and starting from GUI - would you still need to enter PIN?

No.  Thanks for pointing this out.

**Comment by Olf0** [comment]
I just realised that it makes sense to keep this issue open until the original issue ("point 0") is fixed.

**Comment by Olf0** [comment]
Original issue ("point 0") is fixed by commit #114.

**Comment by Olf0** [comment]
I just noticed, that Storeman always uses `X-Nemo-Application-Type=silica-qt5` in its `.desktop` file, as one can see for example in [its `.desktop`  file for SFOS 3.1.x. and 3.2.x](https://github.com/storeman-developers/harbour-storeman/blob/sfos3.2/harbour-storeman.desktop#L3).  Earlier (I traced this back to July 2018), [Storeman also used `Exec=invoker --type=silica-qt5 -s harbour-storeman`](https://github.com/storeman-developers/harbour-storeman/blob/cc39f2d52ba8916086e65b38d85aeb03468a29b7/harbour-storeman.desktop#L5) in addition to `X-Nemo-Application-Type=silica-qt5` (it now uses simply `Exec=harbour-storeman`).

Because Storeman does work fine with `Type=silica-qt5` for all SFOS ≥ 3.1.0, I wonder if the PRs #92 and #114 (addressing issues #82 and #111 "point 0") were merely workarounds for an issue in `sailfishos-chum-gui`'s code, which might be indicated by the `[W] unknown:479 - file:///usr/lib/qt5/qml/Sailfish/Silica/private/PulleyMenuBase.qml:479:13: QML State: Binding loop detected for property "when"` (tested with `sailfishos-chum-gui-0.4.1` on SFOS 4.0.1).
